test/e2e_node/device_plugin.go

```diff
@@ -135,13 +154,20 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
 				return false
 			}, 5*time.Minute, framework.Poll).Should(BeTrue())
 
-			By("Re-Register resources")
-			dp1 = dm.NewDevicePluginStub(devs, socketPath, resourceName, false)
-			dp1.SetAllocFunc(stubAllocFunc)
-			err = dp1.Start()
+			By("Re-Register resources and deleting the pods and waiting for container removal")
+			getOptions := metav1.GetOptions{}
+			gp := int64(0)
+			deleteOptions := metav1.DeleteOptions{
+				GracePeriodSeconds: &gp,
+			}
+			err = f.ClientSet.CoreV1().Pods(metav1.NamespaceSystem).Delete(dp.Name, &deleteOptions)
 			framework.ExpectNoError(err)
+			waitForContainerRemoval(devicePluginPod.Spec.Containers[0].Name, devicePluginPod.Name, devicePluginPod.Namespace)
+			_, err = f.ClientSet.CoreV1().Pods(metav1.NamespaceSystem).Get(dp.Name, getOptions)
```

Comment:
We can use `framework.ExpectNoError(err)` instead of `Expect(err).NotTo(BeNil())`